graphsdk/src/main/java/com/microsoft/graph/model/Date.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.graph.model;
+
+
+import java.util.Locale;
+
+/**
+ * A timezone-nonspecific date
+ */
+public class Date {
+
+    /**
+     * Array indices for date parsing
+     */
+    private static final int
+            sYear = 0,
+            sMonth = 1,
+            sDay = 2;
+
+    /**
+     * The raw string source of this object
+     */
+    private final String rawDateString;
```

Comment:
We should not hold onto the raw string, this class should just be the data field